autoscaler/controllers/actions/tailsampling_controller.go

```diff
@@ -0,0 +1,316 @@
+	}
+
+	conf := sampling.TailSamplingConfig{
+		Policies: actionsPolicies,
+	}
+	configJson, err := json.Marshal(conf)
+	if err != nil {
+		return err
+	}
+
+	processor := &v1.Processor{
+		TypeMeta: metav1.TypeMeta{APIVersion: "odigos.io/v1alpha1", Kind: "Processor"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "tail-sampling-processor",
+			Namespace:       namespace,
+			OwnerReferences: actionsReferences,
+		},
+		Spec: v1.ProcessorSpec{
+			Type:            "tail_sampling",
+			ProcessorName:   "tail_sampling",
+			Disabled:        false, // In case related actions are disabled, the processor wont be created
+			Signals:         []common.ObservabilitySignal{common.TracesObservabilitySignal},
+			CollectorRoles:  []v1.CollectorsGroupRole{v1.CollectorsGroupRoleClusterGateway},
+			OrderHint:       1,
+			ProcessorConfig: runtime.RawExtension{Raw: configJson},
+		},
+	}
+
+	if err := r.Patch(ctx, processor, client.Apply, client.FieldOwner("tail-sampling-controller"), client.ForceOwnership); err != nil {
+		return err
+	}
+
+	// Delete ProbabilisticSampler processor to avoid cases where probabilistic sampler processor already exists
+	return commonconf.DeleteProcessorByType(r.Client, "probabilistic_sampler", namespace)
+}
+
+func (r *TailSamplingReconciler) ReportReconciledToProcessorFailed(ctx context.Context, action metav1.Object, reason, msg string) error {
+	conditions, err := getConditions(action)
+	if err != nil {
+		return err
+	}
+
+	changed := meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               "ActionTransformedToProcessorType",
+		Status:             metav1.ConditionFalse,
+		Reason:             reason,
+		Message:            msg,
+		ObservedGeneration: action.GetGeneration(),
+	})
+
+	if changed {
+		return r.Status().Update(ctx, action.(client.Object))
+	}
+	return nil
+}
+
+func (r *TailSamplingReconciler) ReportReconciledToProcessor(ctx context.Context, action metav1.Object) error {
+	conditions, err := getConditions(action)
+	if err != nil {
+		return err
+	}
+
+	changed := meta.SetStatusCondition(conditions, metav1.Condition{
+		Type:               "ActionTransformedToProcessorType",
+		Status:             metav1.ConditionTrue,
+		Reason:             "ProcessorCreatedReason",
+		Message:            "The action has been reconciled to a processor resource.",
+		ObservedGeneration: action.GetGeneration(),
+	})
+
+	if changed {
+		if err := r.Status().Update(ctx, action.(client.Object)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// In case tail sampling related actions exists, append it to tail_sampling processor
+// otherwise, create a dedicated probablistic sampler processor
+func (r *TailSamplingReconciler) createDedicatedProbabilisticProcessor(relevantActions map[reflect.Type][]metav1.Object) bool {
+	typesLength := make(map[reflect.Type]int)
+
+	for actionType, actions := range relevantActions {
+		typesLength[actionType] = len(actions)
+	}
+
+	allOthersZero := true
```

Comment:
before getting into this function i've added:
```	
if r.isNoRelevantActions(desiredActions) {
		return r.deleteSamplingProcessorsIfExists(ctx, req.Namespace)
	}
```
it helps me remove the cleanups in case no relevant actions in both sync functions + avoid cases as you described.